### DISCUSSION TITLE
SSH key permission errors

### DISCUSSION THREAD

**Discussion by xxComatoastxx** [community/community#42239] [New to GitHub]
### Select Topic Area

Product Feedback

### Body

Hello, I recently join github and succesfully encorperated my ssh keys into my profile. 
Everything was working fine until I got home from work last night and I tried to do a push request on my webpage and I got the following error:

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@         WARNING: UNPROTECTED PRIVATE KEY FILE!          @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
Permissions 0644 for '/path/to/ssh/key' are too open.
It is required that your private key files are NOT accessible by others.
This private key will be ignored.
Load key "/path/to/ssh/key": bad permissions
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

And now I Cannot update my repos and haven't been able since. Can some one help me out?

**Comment by byrneh** [comment]
Sounds like you have accidentally changed your permissions on you ssh key and/or ssh folder to something too permissive if you change back it should start working again. The permissions need to be restrictive to only that user not anh logged in user on a device.

**Reply by airtower-luna** [reply]
One addition: If there's any chance anyone else had access to the machine meanwhile (not counting the admin, who can override file permissions anyway), you should remove the key from your GitHub account (and any other places where you use it) instead, and create a new one. If anyone was able to copy the key they will be able to log in as you now.

**Reply by xxComatoastxx** [reply]
This is actually the second pair of keys I have been through that this has happened with. They work for a day or so and then the issue pops up again and I am certain no one has had access to my keys but me. I live alone and unless the feds are watching my github where I manage my blog it seems unlikely lol. I Am sure I have no changed any permissions. What permission level should work?

**Reply by xxComatoastxx** [reply]
PS: Thank you for the feed back.

**Comment by airtower-luna** [comment]
Only your user should have access, you can remove access for "group" (`g`) and "other" (`o`) with `chmod go= /path/to/ssh/key`.

However, if the permissions change seemingly on their own after a short while there is *something* wrong with the system. Obviously I can't say what without having seen it, but the first thing I'd check are any regular jobs (cron, systemd timers, etc.) that affect the directory in question.
